.github/workflows/Build.yml

```diff
@@ -0,0 +1,63 @@
+name: Run build
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+
+jobs:
+  test:
+    name: "Build (${{ matrix.os }})"
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest]
+
+    steps:
+      - name: "Checking changed files"
+        id: files
+        uses: softprops/diffset@3aeca0719d5043522e155012d3c9f7470fdd2dc6 # v1
+        if: github.event_name == 'pull_request' && github.event.pull_request.head.repo.full_name == github.repository
+        with:
+          base: main
+          global_files: |
+            !{src/**/*}
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      - name: "Checking-out code"
+        uses: actions/checkout@ec3a7ce113134d7a93b817d10a8272cb61118579 # v2
+        with:
+          fetch-depth: 0
+      - name: "Fetching main to compare"
+        if: github.event_name == 'pull_request' && steps.files.outputs.global_files == ''
+        run: git fetch --no-tags --prune --depth=1 origin +refs/heads/${{ github.base_ref }}:refs/remotes/origin/${{ github.base_ref }}
+      - name: "Defining environment variables"
+        if: startsWith(matrix.os, 'ubuntu')
+        id: variablesLinux
+        run: echo "::set-output name=arg::$(if [ "${{ steps.files.outputs.global_files }}" = "" ] && ${{ github.event_name == 'pull_request' }}; then echo '--since'; else echo ''; fi)"
```

Comment:
`set-output` expression is deprecated, I would switch to new syntax 
https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/